packages/modules/imodel-browser/src/containers/ITwinGrid/useITwinFavorites.ts

```diff
@@ -0,0 +1,177 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useState } from "react";
+
+import { ApiOverrides, ITwinFull } from "../../types";
+import { _getAPIServer } from "../../utils/_apiOverrides";
+
+const HOOK_ABORT_ERROR =
+  "The fetch request was aborted by the cleanup function.";
+
+/**
+ * Custom hook to manage iTwin favorites.
+ * @param {string | (() => Promise<string>) | undefined} accessToken - Access token that requires the `itwins:read` scope. Provide a function that returns the token to prevent the token from expiring.
```

Comment:
should just need the `itwin-platform` scope

Reply:
should I also update the scope in preview.js to itwin-platform?

Reply:
yes